infra/cloud_function.tf

```diff
@@ -79,7 +79,7 @@ resource "google_storage_bucket" "relay_function" {
 
 # Upload the Cloud Function source code to the Storage Bucket
 resource "google_storage_bucket_object" "source_code" {
-  name   = "function-source-${terraform.workspace}-${data.archive_file.function_source.output_sha256}.zip"
+  name   = "function-source-${terraform.workspace}-${data.archive_file.function_source.output_md5}.zip"
```

Comment:
according to some stackoverflow x AI recommendations this could reduce terraform's trigger finger, meaning `tf apply` re-deploying the cloud function even though no source code changed